application/slack/v0/task_functions.go

```diff
@@ -81,6 +117,40 @@ func (e *execution) readMessage(in *structpb.Struct) (*structpb.Struct, error) {
 		readTaskResp.Conversations = []Conversation{}
 	}
 
+	// To reduce API calls, we get all user information in one call.
+	var userIDs []string
+	for _, conversation := range readTaskResp.Conversations {
+		userIDs = append(userIDs, conversation.UserID)
+		if len(conversation.ThreadReplyMessage) > 0 {
+			for _, threadReply := range conversation.ThreadReplyMessage {
+				userIDs = append(userIDs, threadReply.UserID)
+			}
+		}
+	}
+
+	userIDs = removeDuplicateUserIDs(userIDs)
+	users, err := e.client.GetUsersInfo(userIDs...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for i, conversation := range readTaskResp.Conversations {
+		for _, user := range *users {
```

Comment:
We can create a `map[string]string` to convert id to name, which will speed things up.